informer/disk/disk.go

```diff
@@ -12,23 +13,61 @@ import (
 	"github.com/ipfs/ipfs-cluster/api"
 )
 
+type MetricType int
+
+const (
+	MetricFreeSpace = iota
+	MetricRepoSize
+)
+
+const DefaultMetric = MetricFreeSpace
+
 var logger = logging.Logger("diskinfo")
 
 // MetricTTL specifies how long our reported metric is valid in seconds.
 var MetricTTL = 30
 
-// MetricName specifies the name of our metric
-var MetricName = "disk"
+// metricToRPC maps from a specified metric name to the corrresponding RPC call
+var metricToRPC = map[MetricType]string{
+	MetricFreeSpace: "IPFSFreeSpace",
+	MetricRepoSize:  "IPFSRepoSize",
+}
 
 // Informer is a simple object to implement the ipfscluster.Informer
 // and Component interfaces.
 type Informer struct {
+	Type      MetricType
+	name      string
 	rpcClient *rpc.Client
+	rpcName   string
 }
 
-// NewInformer returns an initialized Informer.
-func NewInformer() *Informer {
-	return &Informer{}
+// NewInformer returns an initialized Informer that uses the DefaultMetric. The
+// name argument is meant as a user-facing identifier for the Informer and can
+// be anything.
+func NewInformer(name string) *Informer {
+	return &Informer{
+		name:    name,
+		rpcName: metricToRPC[DefaultMetric],
+	}
+}
+
+// NewInformerWithMetric returns an Informer that uses the input MetricType. The
+// name argument has the same purpose as in NewInformer.
+func NewInformerWithMetric(metric MetricType, name string) *Informer {
```

Comment:
Can we return (*Informer, error) and set the error of the metric type is invalid here?

Reply:
Ah, right, good idea. Definitely want to do that.